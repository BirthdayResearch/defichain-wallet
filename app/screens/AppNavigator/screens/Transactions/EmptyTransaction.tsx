import * as React from 'react'
import { View } from 'react-native'
import { NavigationProp } from '@react-navigation/native'
import { MaterialCommunityIcons } from '@expo/vector-icons'
import { TransactionsParamList } from './TransactionsNavigator'
import { tailwind } from '../../../../tailwind'
import { Text } from '../../../../components'
import { translate } from '../../../../translations'
import { Button } from '../../../../components/Button'

interface EmptyTransactionProps {
  navigation: NavigationProp<TransactionsParamList>
}

export function EmptyTransaction (props: EmptyTransactionProps): JSX.Element {
  return (
    <View
      testID='empty_transaction'
      style={tailwind('px-8 pt-32 pb-2 text-center')}
    >
      <MaterialCommunityIcons name='folder-alert' size={44} style={tailwind('pb-5 text-center text-black')} />
      <Text style={tailwind('text-2xl pb-2 font-semibold text-center')}>
        {translate('screens/TransactionsScreen', 'No transactions yet')}
      </Text>
      <Text style={tailwind('text-sm pb-16 text-center opacity-60')}>
        {translate('screens/TransactionsScreen', 'Start transacting with your wallet. All transactions made will be displayed here.')}
      </Text>
      <Button
        testID='button_receive_coins'
        title='Receive Coins'
        onPress={() => props.navigation.navigate('Receive')}
        label={translate('screens/TransactionsScreen', 'RECEIVE COINS')}
      />
    </View>
  )
}
