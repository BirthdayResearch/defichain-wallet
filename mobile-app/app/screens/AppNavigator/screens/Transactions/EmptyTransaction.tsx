import { NavigationProp } from '@react-navigation/native'
import { RefreshControl } from 'react-native'
import { Button } from '@components/Button'
import { ThemedIcon, ThemedScrollView, ThemedText } from '@components/themed'
import { tailwind } from '@tailwind'
import { translate } from '@translations'
import { TransactionsParamList } from './TransactionsNavigator'

interface EmptyTransactionProps {
  navigation: NavigationProp<TransactionsParamList>
  handleRefresh: (nextToken?: string | undefined) => void
  loadingStatus: string
}

export function EmptyTransaction (props: EmptyTransactionProps): JSX.Element {
  return (
    <ThemedScrollView
      refreshControl={
        <RefreshControl
          onRefresh={props.handleRefresh}
          refreshing={props.loadingStatus === 'loading'}
        />
      }
      style={tailwind('px-8 pt-32 pb-2 text-center')}
      testID='empty_transaction'
    >
      <ThemedIcon
        dark={tailwind('text-white')}
        iconType='MaterialIcons'
        light={tailwind('text-black')}
        name='assignment-late'
        size={44}
        style={tailwind('pb-5 text-center')}
      />

      <ThemedText style={tailwind('text-2xl pb-2 font-semibold text-center')}>
        {translate('screens/TransactionsScreen', 'No transactions yet')}
      </ThemedText>

      <ThemedText style={tailwind('text-sm pb-16 text-center opacity-60')}>
        {translate('screens/TransactionsScreen', 'Start transacting with your wallet. All UTXO transactions made will be displayed here. Other transaction types are not supported yet.')}
      </ThemedText>

      <Button
        label={translate('screens/TransactionsScreen', 'RECEIVE TOKENS')}
        onPress={() => props.navigation.navigate('Receive')}
        testID='button_receive_coins'
        title='Receive Tokens'
      />
    </ThemedScrollView>
  )
}
