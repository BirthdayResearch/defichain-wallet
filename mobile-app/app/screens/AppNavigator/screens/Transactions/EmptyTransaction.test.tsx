import { fireEvent, render } from '@testing-library/react-native'
import { EmptyTransaction } from './EmptyTransaction'

jest.mock('@react-navigation/native')
jest.mock('@shared-contexts/ThemeProvider')

describe('empty transaction', () => {
  it('should match snapshot', async () => {
    const navigation: any = {
      navigate: jest.fn()
    }
    const rendered = render(<EmptyTransaction
      handleRefresh={() => {
      }}
      key='1'
      loadingStatus='loading'
      navigation={navigation}
                            />)
    expect(rendered.toJSON()).toMatchSnapshot()
    const receiveButton = await rendered.findByTestId('button_receive_coins')

    const spy = jest.spyOn(navigation, 'navigate')
    fireEvent.press(receiveButton)
    expect(spy).toHaveBeenCalled()
  })
})
