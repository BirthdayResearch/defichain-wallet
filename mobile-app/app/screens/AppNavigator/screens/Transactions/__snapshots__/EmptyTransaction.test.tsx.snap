// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty transaction should match snapshot 1`] = `
<RCTScrollView
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={true}
    />
  }
  style={
    Array [
      Object {
        "paddingBottom": 8,
        "paddingLeft": 32,
        "paddingRight": 32,
        "paddingTop": 128,
        "textAlign": "center",
      },
      Object {
        "backgroundColor": "rgba(245, 245, 245, 1)",
      },
    ]
  }
  testID="empty_transaction"
>
  <RCTRefreshControl />
  <View>
    <Text />
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontFamily": "SemiBoldFont",
              "fontSize": 24,
              "fontWeight": "600",
              "lineHeight": 32,
              "paddingBottom": 8,
              "textAlign": "center",
            },
            Object {
              "color": "rgba(0, 0, 0, 1)",
            },
          ],
        ]
      }
    >
      No transactions yet
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "RegularFont",
            "fontSize": 16,
            "fontWeight": "400",
            "lineHeight": 24,
          },
          Array [
            Object {
              "fontSize": 14,
              "lineHeight": 20,
              "opacity": 0.6,
              "paddingBottom": 64,
              "textAlign": "center",
            },
            Object {
              "color": "rgba(0, 0, 0, 1)",
            },
          ],
        ]
      }
    >
      Start transacting with your wallet. All UTXO transactions made will be displayed here. Other transaction types are not supported yet.
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(255, 229, 247, 1)",
          "borderBottomLeftRadius": 4,
          "borderBottomRightRadius": 4,
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 16,
          "marginLeft": 16,
          "marginRight": 16,
          "marginTop": 32,
          "opacity": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        }
      }
      testID="button_receive_coins"
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "RegularFont",
              "fontSize": 16,
              "fontWeight": "400",
              "lineHeight": 24,
            },
            Object {
              "color": "rgba(255, 0, 175, 1)",
              "fontFamily": "BoldFont",
              "fontWeight": "700",
              "textAlign": "center",
            },
          ]
        }
      >
        RECEIVE TOKENS
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
